app/services/authorizer.rb

```diff
@@ -104,10 +104,18 @@ def build_filtered_scope_components(resource_class, all_filters, options)
     return result if all_filters.any?(&:unlimited?)
 
     search_string = build_scoped_search_condition(all_filters.select(&:limited?))
-    find_options = ScopedSearch::QueryBuilder.build_query(resource_class.scoped_search_definition, search_string, options)
-    result[:where] << find_options[:conditions]
-    result[:includes].push(*find_options[:include])
-    result[:joins].push(*find_options[:joins])
+
+    begin
+      find_options = ScopedSearch::QueryBuilder.build_query(resource_class.scoped_search_definition, search_string, options)
+
+      result[:where] << find_options[:conditions]
+      result[:includes].push(*find_options[:include])
+      result[:joins].push(*find_options[:joins])
+    rescue ScopedSearch::QueryNotSupported => e
+      Foreman::Logging.logger('permissions').error N_("Scoped search query not supported: #{e.message}")
```

Comment:
For Evgeni who had been on the project for almost 10 years and still has no clue: will this provide the error to the user in the UI, or will they just get an empty list and need to go dive into logs to understand whats happened?

Reply:
ok